packages/cache/src/internal/requestUtils.ts

```diff
@@ -111,7 +109,8 @@ export async function retryTypedResponse<T>(
         return {
           statusCode: error.statusCode,
           result: null,
-          headers: {}
+          headers: {},
+          error: error
```

Comment:
There are tests for [requestUtils](https://github.com/actions/toolkit/blob/e779919848e5a3ee655ab825e53f8ac979390a5c/packages/cache/__tests__/requestUtils.test.ts). Update/add tests to include testing `error` property.